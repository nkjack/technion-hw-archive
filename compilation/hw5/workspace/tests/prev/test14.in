byte returnByte() {
	return 200b;
}
void maxTestsAssignFromFunc() {
	byte x = returnByte();
	printi(x);
}


void main() {
     maxTestsAssignFromFunc();
     print("\nI hate this course\n");
}
