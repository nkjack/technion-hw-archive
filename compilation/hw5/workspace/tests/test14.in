
bool isGreater(int x,int y,bool flag)
@pre(flag)
@pre(flag and not not flag and (x >= y))
{
	return true;
}
void TestFunc(bool flag) 
@pre(not flag)
{
	int x=5;
	print("x = ");
	printi(x);
	print("\n");
	{
		int y;
		{
			int z = x;
		}
		y = x*x;

		while(true) {
			print("y = ");
			printi(y);
			print("\n");
			if(not (y<1) and isGreater(y,x,not flag)){
					y =y - x;
					continue;
			}
			else
				break;
		}
		
		print("y reached 0\n");
				
	}
}

void main() {
	TestFunc(false);
}