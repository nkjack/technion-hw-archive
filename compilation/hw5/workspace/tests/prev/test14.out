Loaded: ./exceptions.s
200
I hate this course
